src/components/Scenarios/Scenarios.jsx

```diff
@@ -0,0 +1,227 @@
+    const { dropDownValue, formFields } = this.state;
+    const { data } = scenariosState[dropDownValue];
+    const { metadata, metadataLoading, metadataError } = scenariosState.metadata;
+    return (
+      <StyledDiv>
+        {metadataError && <Error />}
+        {metadataLoading && <Spinner />}
+        {!metadataError && !metadataLoading &&
+        <div>
+          <Heading title={strings.header_scenarios} subtitle={strings.scenarios_subtitle} />
+          <Tabs value={dropDownValue} onChange={this.handleChange} style={tabsStyle}>
+            {menuItems.map(item => (
+              <Tab label={item.text} value={item.value} icon={item.icon} containerElement={getLink(item.value)} className="tab" />
+            ))}
+          </Tabs>
+          <div style={formFieldStyle}>
+            {fields[dropDownValue].map(field => (
+              <ScenariosFormField
+                key={field + dropDownValue}
+                field={field}
+                updateQueryParams={this.updateQueryParams}
+                updateFormFieldState={this.updateFormFieldStates}
+                formFieldState={formFields[dropDownValue] && formFields[dropDownValue][field]}
+                metadata={metadata}
+              />
+          ))}
+          </div>
+          <FlatButton
+            onClick={this.getData}
+            style={buttonStyle}
+            backgroundColor="rgba(220, 220, 220, 0.05)"
+            hoverColor="rgba(220, 220, 220, 0.2)"
+            label={strings.explorer_query_button}
+            icon={<ActionSearch />}
+            primary
+          />
+          <Heading title={strings.explorer_results} subtitle={`${data.filter(minSampleSize).length} ${strings.explorer_num_rows}`} />
+          <Table
+            key={dropDownValue}
+            data={data.filter(minSampleSize)}
+            columns={getColumns(dropDownValue, metadata)}
+            loading={scenariosState[dropDownValue].loading}
+            paginated
+          />
+        </div>
+        }
+      </StyledDiv>
+    );
+  }
+}
+
+const mapStateToProps = (state) => {
+  const {
+    scenariosItemTimings,
+    scenariosLaneRoles,
+    scenariosMisc,
+    metadata,
+  } = state.app;
+
+  return {
+    scenariosState: {
+      metadata: {
+        metadata: metadata.data.scenarios,
+        metadataLoading: metadata.loading,
+        metadataError: metadata.error,
+      },
+      itemTimings: {
+        data: scenariosItemTimings.data,
+        loading: scenariosItemTimings.loading,
+        error: scenariosItemTimings.error,
+      },
+      laneRoles: {
+        data: scenariosLaneRoles.data,
+        loading: scenariosLaneRoles.loading,
+        error: scenariosLaneRoles.error,
+      },
+      misc: {
+        data: reduceRows(groupByArray(scenariosMisc.data, 'scenario')),
+        loading: scenariosMisc.loading,
+        error: scenariosMisc.error,
+      },
+    },
+  };
+};
+
+const mapDispatchToProps = dispatch => ({
+  scenariosDispatch: {
```

Comment:
don't know if this nesting is necessary, i think you can just put these props in directly

Reply:
I put them inside an object so I can call the functions dynamically. not sure if there is another way to do it.

Reply:
nevermind I changed it